src/Schema/Builder.php

```diff
@@ -165,9 +171,23 @@ public function getTables()
         return $collections;
     }
 
-    public function getTableListing()
+    public function getTableListing($schema = null, $schemaQualified = true)
     {
-        $collections = iterator_to_array($this->connection->getMongoDB()->listCollectionNames());
+        $collections = [];
+
+        if ($schema === null || is_string($schema)) {
+            $collections[$schema ?? ''] = iterator_to_array($this->connection->getDatabase($schema)->listCollectionNames());
+        } elseif (is_array($schema)) {
+            foreach ($schema as $db) {
+                $collections[$db] = iterator_to_array($this->connection->getDatabase($db)->listCollectionNames());
+            }
+        }
+
+        if ($schema && $schemaQualified) {
+            $collections = array_map(fn ($db, $collections) => array_map(static fn ($collection) => $db . '.' . $collection, $collections), array_keys($collections), $collections);
+        }
+
+        $collections = array_merge(...array_values($collections));
```

Comment:
If `$schemaQualified` is false and the database name is never prepended to yield full namespaces, would this statement potentially clobber collection names?

Reply:
Yes, but that's how the feature is designed.